Src/PerformanceCollector/Shared/Implementation/QuickPulse/Process/PerfLib/CategorySample.cs

```diff
@@ -0,0 +1,258 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.PerfLib
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Runtime.InteropServices;
+
+    internal class CategorySample
+    {
+        public Dictionary<int, CounterDefinitionSample> CounterTable;
```

Comment:
public CategorySample(Dictionary<int, CounterDefinitionSample> counterTable, ...) { ... }

class CategorySampleMock : CategorySample {
    public CategorySampleMock() : base(whateverDictionaryYouWantInTheMock) { ... }
...
}

You get the idea.